src/tools.js

```diff
@@ -636,46 +640,59 @@ window.tools = [{
 	},
 	paint(ctx, x, y) {
 		if(this.points.length < 1){ return; }
-		
+
 		const i = this.points.length - 1;
 		this.points[i].x = x;
 		this.points[i].y = y;
-	},
-	draw_curve(ctx, x, y) {
-		if(this.points.length < 1){ return; }
 		
-		update_brush_for_drawing_lines(stroke_size);
-		
-		if(this.points.length === 4){
+		this.preview_canvas.ctx.clearRect(0, 0, this.preview_canvas.width, this.preview_canvas.height);
+		this.preview_canvas.ctx.strokeStyle = stroke_color;
+
+		// Draw curves on preview canvas
+		if (this.points.length === 4) {
 			draw_bezier_curve(
-				ctx,
+				this.preview_canvas.ctx,
 				this.points[0].x, this.points[0].y,
 				this.points[2].x, this.points[2].y,
 				this.points[3].x, this.points[3].y,
 				this.points[1].x, this.points[1].y,
 				stroke_size
 			);
-		}else if(this.points.length === 3){
+		}
+		else if (this.points.length === 3) {
 			draw_quadratic_curve(
-				ctx,
+				this.preview_canvas.ctx,
 				this.points[0].x, this.points[0].y,
 				this.points[2].x, this.points[2].y,
 				this.points[1].x, this.points[1].y,
 				stroke_size
 			);
-		}else{
-			draw_line(
-				ctx,
-				this.points[0].x, this.points[0].y,
-				this.points[1].x, this.points[1].y,
-				stroke_size
+		}
+		else {
+			draw_line_strip(
+				this.preview_canvas.ctx,
+				this.points
 			);
 		}
+		
+	},
+	draw_curve(ctx, x, y) {
```

Comment:
`draw_curve` is no longer a particularly meaningful name. It doesn't render the curve, and it doesn't really care what's on the preview canvas.
It's also no longer a substantial function, and I would just inline it in the two places it's used.